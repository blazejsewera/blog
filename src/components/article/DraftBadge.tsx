import type { Component } from 'solid-js'
import { cx } from '../util/classes'

export const DraftBadge: Component = () => (
  <div
    class={cx([
      'inline-flex',
      'rounded-full',
      'border-2',
      'w-min',
      'h-min',
      'px-3',
      'py-0.5',
      'uppercase',
      'text-xs',
      'border-orange-600',
      'text-orange-600',
      'bg-orange-100',
    ])}
  >
    <div class="m-auto">draft</div>
  </div>
)
