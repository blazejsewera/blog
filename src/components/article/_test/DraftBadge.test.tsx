import { render } from '@solidjs/testing-library'
import { DraftBadge } from '../DraftBadge'

describe('<DraftBadge />', () => {
  const { container, unmount } = render(() => <DraftBadge />)

  expect(container).toMatchSnapshot()

  unmount()
})
