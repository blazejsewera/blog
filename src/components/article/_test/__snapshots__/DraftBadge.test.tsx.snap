// Vitest Snapshot v1

exports[`<DraftBadge /> > renders properly 1`] = `
<div>
  <div
    class="inline-flex rounded-full border-2 w-min px-3 uppercase text-xs border-orange-600 text-orange-600 bg-orange-100"
  >
    <div
      class="m-auto"
    >
      draft
    </div>
  </div>
</div>
`;
